src/http/index.ts

```diff
@@ -17,20 +17,26 @@ const logger = getLogger('streamr:httpAdapter')
 
 export const start = (
     { port, privateKeyFileName, certFileName }: AdapterConfig, 
-    { networkNode, publisher, streamFetcher, metricsContext, cassandraStorage, storageConfig}: BrokerUtils
+    { config, networkNode, publisher, streamFetcher, metricsContext, cassandraStorage, storageConfig}: BrokerUtils
 ) => {
     const app = express()
 
     // Add CORS headers
     app.use(cors())
 
     // Rest endpoints
-    app.use('/api/v1', dataQueryEndpoints(networkNode, streamFetcher, metricsContext))
     app.use('/api/v1', dataProduceEndpoints(streamFetcher, publisher))
     app.use('/api/v1', volumeEndpoint(metricsContext))
 
-    app.use('/api/v1', dataMetadataEndpoint(cassandraStorage))
-    app.use('/api/v1', storageConfigEndpoints(storageConfig))
+    if (config.network.isStorageNode) {
+        app.use('/api/v1', dataQueryEndpoints(networkNode, streamFetcher, metricsContext))
+        app.use('/api/v1', dataMetadataEndpoint(cassandraStorage))
+        app.use('/api/v1', storageConfigEndpoints(storageConfig))    
+    }
+
+    app.use('*', (_req: Request, res: Response) => {
```

Comment:
Would `404` be the appropriate status code in this case? I was thinking `501`  being more of an status response code for a known but "deactivated" HTTP endpoint.

Reply:
Yes, it would be a valid status code. 

But if we use that, there would be same status codes for a valid route (e.g. `/streams/:id/storage/partitions/:partition` response of "not configured"), and an invalid route (e.g. `/streams/:id/**strage**/partitions/:partition`).

Also either HTTP adapter would need to know about the deactivated endpoints, or each endpoint definition would need to check that independently. This model where HTTP adapter just checks if some endpoints are needed or not, is more straightforward.